tests/topotests/lib/common_config.py

```diff
@@ -4995,3 +4997,64 @@ def scapy_send_raw_packet(tgen, topo, senderRouter, intf, packet=None):
 
     logger.debug("Exiting lib API: {}".format(sys._getframe().f_code.co_name))
     return True
+
+
+def verify_kernel_ping(tgen, router, address, count=None, addrtype="ipv4", intf=None):
+    """
+    API to ping from kernel and verify the result
+
+    Packages used : PingParsing
+
+    Parameters
+    ----------
+    * `tgen`: topogen object
+    * `router`: router on which ping has to be performed
+    * `address` : ping destination address
+    * `count` : number of ping packets to be sent.
+    * `addr_type` : ping ipv4 or ipv6 addresses.
+    * `intf` : ping using interface as out interface.
+
+    Usage
+    -----
+    result = verify_kernel_ping(tgen, 'r1', '1.1.1.1', '1')
+
+    Returns
+    -------
+    1) True for successful ping
+    2) errormsg - when ping fails
+    """
+
+    logger.debug("Entering lib API: verify_kernel_ping()")
+    rnode = tgen.routers()[router]
+
+    if not address:
+        errormsg = "Destination address to be pinged is missing."
+        return errormsg
+
+    if addrtype is "ipv6":
+        ping_cmd = "ping6 {}".format(str(address))
+    else:
+        ping_cmd = "ping {}".format(str(address))
+
+    # interface
+    if intf:
+        ping_cmd += " -I {}".format(str(intf))
+
+    # count
+    if count:
+        ping_cmd += " -c {}".format(str(count))
+
+    logger.info("On {} running ping command: [{}]".format(router, ping_cmd))
+    parser = pingparsing.PingParsing()
+    pingresult = rnode.run(ping_cmd)
+    counter = -1
+    if parser.parse(pingresult):
+        counter = parser.parse(pingresult).as_dict()["packet_loss_count"]
+
+    if counter is not 0:
```

Comment:
It seems rather excessive to go parsing the `ping` output just to check that the packet loss is 0 — an exit code check should suffice for that. Can you either provide some additional justification for the parser, or remove the extra steps?

Reply:
hey @eqvinox
 The intention of metioned method  was to parse the ping output into python dict and based on the loss count I was validating the  ping  operation
anyways , I had implemented in a simpler way .
check now